_tools/chroma2highcontrastchroma.R

```diff
@@ -0,0 +1,222 @@
+library("magrittr")
+
+# Find color passing --------------------
+
+get_letter <- function(rgb, letter) {
+  if (rgb@coords[1, letter] <= 0.03928) {
+    return(rgb@coords[1, letter]/12.92)
+  } else {
+    return(((rgb@coords[1, letter] +0.055)/1.055) ^ 2.4)
+  }
+}
+
+.calculate_luminance <- function(hex) {
+  # https://www.w3.org/TR/WCAG20-TECHS/G17.html
+  
+  rgb <- colorspace::hex2RGB(hex)
+  
+  R <- get_letter(rgb, "R")
+  G <- get_letter(rgb, "G")
+  B <- get_letter(rgb, "B")
+  
+  0.2126 * R + 0.7152 * G + 0.0722 * B
+}
+
+calculate_luminance <- memoise::memoise(.calculate_luminance)
+
+calculate_contrast <- function(color, background) {
+  # https://www.w3.org/TR/WCAG20-TECHS/G17.html
+  l1 <- calculate_luminance(color)
+  l2 <- calculate_luminance(background)
+  
+  
+  (max(l1, l2) + 0.05) / (min(l1, l2) + 0.05)
+  
+
+}
+
+
+lengthen_color <- function(color) {
+  if (stringr::str_length(color) == 7) {
+    return(color)
+  } else {
+    color <- glue::glue_collapse(c("#",
+                                   rep(substring(color, 2, 2), 2),
+                                   rep(substring(color, 3, 3), 2),
+                                   rep(substring(color, 4, 4), 2)))
+    
+    return(color)
+  }
+  
+}
+
+replace_color <- function(original_color, background, wcag, verbose = FALSE) {
+
+  if(wcag == "AAA") {
+    threshold <- 7
+  } else{
+    threshold <- 4.5
+  }
+  
+  original_color <- lengthen_color(original_color)
+  background <- lengthen_color(background)
+  
+  color <- original_color
+
+  while (calculate_contrast(color, background) < threshold) {
+    if(verbose) {
+      message(
+        glue::glue(
+          "still searching a replacement for {color} on {background}"
+          )
+        )
+    }
+    
+    if (calculate_luminance(original_color) > calculate_luminance(background)) {
+      if (calculate_luminance(color) == 1) {
+        message(glue::glue("We cannot do better than {color} for {original_color} on {background}"))
```

Comment:
I only get this message for styles/rainbow_dash.go where there's

```go
	chroma.Error:               "bg:#cc0000 #ffffff",
```

In that case it might be better to change the background.